src/agents/der_aggregator.jl

```diff
@@ -0,0 +1,636 @@
+        (der_aggregator.dera_stor_incentive_function[i+1, "incentive"] - der_aggregator.dera_stor_incentive_function[i, "incentive"])
+
+        incentive_function_intercept[i] = der_aggregator.dera_stor_incentive_function[i+1, "participation"] - incentive_function_slope[i] * der_aggregator.dera_stor_incentive_function[i+1, "incentive"]
+    end
+    incentive_level_by_segment = Dict(z => zeros(incentive_function_dimension - 1) for z in model_data.index_z)
+    participation_by_segment = Dict(z => zeros(incentive_function_dimension - 1) for z in model_data.index_z)
+    obj_by_segment = Dict(z => zeros(incentive_function_dimension - 1) for z in model_data.index_z)
+
+    total_der_stor_capacity = make_keyed_array(model_data.index_z, model_data.index_h)
+    for z in model_data.index_z, h in model_data.index_h
+        if w_iter >= 2
+            total_der_stor_capacity(z, h, :) .=
+                customers.x_DG_E_my(reg_year_index, h, z, :BTMStorage) + sum(
+                    customers.x_DG_new_my(Symbol(Int(y)), h, z, :BTMStorage) for
+                    y in model_data.year(first(model_data.index_y_fix)):(reg_year)
+                )
+        else
+            total_der_stor_capacity(z, h, :) .= customers.x_DG_E_my(reg_year_index, h, z, :BTMStorage)
+        end
+    end
+    
+    for z in model_data.index_z
+        for i in 1:incentive_function_dimension - 1
+            # x (incentive) should be $/MW (per year)?
+            DERA_WM = get_new_jump_model(dera_opts.solvers)
+            @variable(DERA_WM, der_aggregator.dera_stor_incentive_function[i, "incentive"] <= incentive <= der_aggregator.dera_stor_incentive_function[i+1, "incentive"])
+            @variable(DERA_WM, dera_charge[model_data.index_d, model_data.index_t] >= 0)
+            @variable(DERA_WM, dera_discharge[model_data.index_d, model_data.index_t] >= 0)
+            @variable(DERA_WM, dera_energy[model_data.index_d, model_data.index_t] >= 0)
+            @variable(DERA_WM, dera_pv_gen[model_data.index_d, model_data.index_t] >= 0)
+
+            dera_stor_capacity_h =
+                h -> begin
+
+                (incentive_function_intercept[i] + incentive_function_slope[i] * incentive) * total_der_stor_capacity(z, h)
+
+            end
+
+            dera_stor_capacity =
+                begin
+
+                sum(dera_stor_capacity_h(h) for h in model_data.index_h)
+
+            end
+
+            # when it comes to the aggregated pv capacity, use aggregated storage capacity, divide by optimal storage size to get the number of households that's participating the dera aggregation,
+            # then multiply by the optimal pv size,
+            # TODO: I'm using optimal storage and pv capacities for existing resources, need to be more flexible to account for changing optimal tech sizes overtime.
+            # right now, assume the optimal sizes are the same for both existing and new techs.
```

Comment:
Could all of this be simpler (not just here, but in other modules as well) if the Customers module tracked capacities differently?